codalab/server/bundle_manager.py

```diff
@@ -332,37 +335,32 @@ def _schedule_run_bundles_on_workers(self, workers, staged_bundles_to_run, user_
 
         # Dispatch bundles
         for bundle, bundle_resources in staged_bundles_to_run:
+            codalab_owned_workers = copy.deepcopy(
+                workers.user_owned_workers(self._model.root_user_id)
+            )
+            user_owned_workers = copy.deepcopy(workers.user_owned_workers(bundle.owner_id))
+            if bundle.metadata.request_queue:
```

Comment:
Can't this logic go in the filter workers?  That would be cleaner, so all this function does is get the set of candidate workers?

Reply:
Moved back.